packages/form-js-editor/src/features/properties-panel/entries/factories/simpleRangeIntegerEntryFactory.js

```diff
@@ -39,31 +40,39 @@ const SimpleRangeIntegerEntry = (props) => {
     path,
     field,
     editField,
-    min,
-    max,
-    defaultValue
+    min = Number.MIN_SAFE_INTEGER,
+    max = Number.MAX_SAFE_INTEGER
   } = props;
 
   const debounce = useService('debounce');
 
   const getValue = () => {
-    const value = get(field, path, defaultValue);
-    return Number.isInteger(value) ? value : defaultValue;
+    const value = get(field, path);
+    const isValid = isValidNumber(value) && Number.isInteger(value);
+    return isValid ? value : null;
   };
 
-  const setValue = (value) => {
-    editField(field, path, value);
+  const setValue = (value, error) => {
+    if (error) {
+      return;
+    }
+
+    editField(field, path, Number(value));
   };
 
-  return NumberFieldEntry({
+  return TextFieldEntry({
     debounce,
     label,
     element: field,
-    step: 1,
-    min,
-    max,
     getValue,
     id,
-    setValue
+    setValue,
+    validate: (value) => {
+      if (value === undefined || value === null || value === '') return;
+      if (!isValidNumber(value)) return 'Should be a valid number.';
+      if (!Number.isInteger(Number(value))) return 'Should be an integer.';
+      if (Big(value).cmp(min) < 0) return `Should be at least ${min}.`;
+      if (Big(value).cmp(max) > 0) return `Should be at most ${max}.`;
```

Comment:
I think it's better to add braces in each if case, so it's more resilient to mistakes if we ever want to change the logic